server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1849,6 +1849,13 @@ private void validateRootVolumeDetachAttach(VolumeVO volume, UserVmVO vm) {
         }
     }
 
+    public Volume detachVolumeViaDestroyVM(long vmId, long volumeId) {
```

Comment:
couple of questions here:
Is this really public? it may a only be called as part of destroy volume!
How different is it really (even being consequential) from regular detachVolumeFromVM? it should at least also fire a detach event to keep usage data in check.